### PR TITLE
Upgraded azure dependencies

### PR DESCRIPTION
**Pull request by evanxg852000**
### Description

Upgraded azure blob storage dependencies

Closes https://github.com/quickwit-oss/quickwit/issues/2053

### CODE REVIEW

File: quickwit/quickwit-storage/src/object_storage/azure_compatible_storage.rs [hunk 1]
```diff
@@ -302,9 +303,14 @@ impl Storage for AzureBlobStorage {
         let mut output_stream = self.container_client.blob_client(name).get().into_stream();
 
         let mut dest_file = File::create(output_path).await?;
-        while let Some(result) = output_stream.next().await {
-            let chunk = result.map_err(AzureErrorWrapper::from)?.data;
-            dest_file.write_all(&chunk).await?;
+        while let Some(chunk_result) = output_stream.next().await {
+            let chunk_response = chunk_result.map_err(AzureErrorWrapper::from)?;
+            let chuck_data = chunk_response
```
Comment (guilload) [inline quickwit/quickwit-storage/src/object_storage/azure_compatible_storage.rs:308]: Do we have to collect here? Can we put a buf writer in front of the file and stream the data into the file directly? Or at least reuse the same vec between chunks?

File: quickwit/quickwit-storage/src/object_storage/azure_compatible_storage.rs [hunk 2]
```diff
@@ -436,8 +442,8 @@ async fn download_all(
         .object_storage_download_num_bytes
         .clone();
     while let Some(chunk_result) = chunk_stream.next().await {
-        let chunk_response = chunk_result.map_err(AzureErrorWrapper::from)?;
-        let chuck_data = chunk_response.data;
+        let chunk_response = chunk_result?;
```
Comment (guilload) [inline quickwit/quickwit-storage/src/object_storage/azure_compatible_storage.rs:445]: Same here.

Comment (guilload) [reply quickwit/quickwit-storage/src/object_storage/azure_compatible_storage.rs:449]: Why do we do `let chunk_response = chunk_result.map_err(AzureErrorWrapper::from)?;` in one case and `let chunk_response = chunk_result?;` in another?

File: quickwit/quickwit-storage/src/object_storage/azure_compatible_storage.rs [hunk 3]
```diff
@@ -436,11 +446,17 @@ async fn download_all(
...
-        object_storage_download_num_bytes.inc_by(chuck_data.len() as u64);
-        output.extend(chuck_data.as_ref());
+        let chunk_response = chunk_result?;
+        let stream_reader = chunk_response
+            .data
+            .map_err(|err| FutureError::new(FutureErrorKind::Other, err))
+            .into_async_read()
+            .compat();
+        let mut body_reader = BufReader::new(stream_reader);
+        let num_byte_copied = tokio::io::copy_buf(&mut body_reader, output).await?;
```
Comment (guilload) [inline quickwit/quickwit-storage/src/object_storage/azure_compatible_storage.rs:456]:
```suggestion
        let num_bytes_copied = tokio::io::copy_buf(&mut body_reader, output).await?;
```

File: quickwit/quickwit-storage/src/object_storage/s3_compatible_storage.rs [hunk 4]
```diff
@@ -457,25 +457,23 @@ impl S3CompatibleObjectStorage {
...
     let object_storage_download_num_bytes = crate::STORAGE_METRICS
         .object_storage_download_num_bytes
         .clone();
-    while let Some(chunk_res) = byte_stream.next().await {
-        let chunk = chunk_res?;
-        object_storage_download_num_bytes.inc_by(chunk.len() as u64);
-        output.extend(chunk.as_ref());
-    }
+    let mut body_reader = BufReader::new(byte_stream.into_async_read());
+    let num_byte_copied = tokio::io::copy(&mut body_reader, output).await?;
```
Comment (guilload) [inline quickwit/quickwit-storage/src/object_storage/s3_compatible_storage.rs:475]:
```suggestion
    let num_bytes_copied = tokio::io::copy(&mut body_reader, output).await?;
```

File: quickwit/quickwit-storage/src/object_storage/azure_compatible_storage.rs (hunk 2 above)
Comment (evanxg852000) [reply quickwit/quickwit-storage/src/object_storage/azure_compatible_storage.rs:449]: - In this case `let chunk_response = chunk_result.map_err(AzureErrorWrapper::from)?;` we need to propagate `StorageError`, and we cannot jump from `azure::Error` to `StrorageError` we need this map. 

In this case `chunk_response = chunk_result?;` we just need to propagate `AzureErrorWrapper`  

do you think removing it by providing an implementation from `azure::Error` to `StrorageError` makes it better?

**Review by guilload — APPROVED** [review]

**Pull request merged**